spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteManifestsAction.java

```diff
@@ -466,6 +471,163 @@ public void testRewriteLargeManifestsPartitionedTable() throws IOException {
+    Integer c3PartitionMin =
+        Conversions.fromByteBuffer(
+            Types.IntegerType.get(), manifests.get(0).partitions().get(2).lowerBound());
+    Integer c3PartitionMax =
+        Conversions.fromByteBuffer(
+            Types.IntegerType.get(), manifests.get(0).partitions().get(2).upperBound());
+
+    // Set the target manifest size to a small value to force splitting records into multiple files
+    table
+        .updateProperties()
+        .set(
+            TableProperties.MANIFEST_TARGET_SIZE_BYTES,
+            String.valueOf(manifests.get(0).length() / 2))
+        .commit();
+
+    SparkActions actions = SparkActions.get();
+
+    List<String> manifestSortKeys = ImmutableList.of("c3_bucket", "c2_trunc", "c1");
+    RewriteManifests.Result result =
+        actions
+            .rewriteManifests(table)
+            .rewriteIf(manifest -> true)
+            .sort(manifestSortKeys)
+            .option(RewriteManifestsSparkAction.USE_CACHING, useCaching)
+            .execute();
+
+    table.refresh();
+    List<ManifestFile> newManifests = table.currentSnapshot().allManifests(table.io());
+
+    assertThat(result.rewrittenManifests()).hasSize(1);
+    assertThat(result.addedManifests()).hasSizeGreaterThanOrEqualTo(2);
+
+    assertThat(newManifests).hasSizeGreaterThanOrEqualTo(2);
+
+    // Rewritten manifests are clustered by c3_bucket - each should contain only a subset of the
+    // lower and upper bounds
+    // of the partition 'c3'.
+    List<Pair<Integer, Integer>> c3Boundaries =
+        newManifests.stream()
+            .map(manifest -> manifest.partitions().get(2))
+            .sorted(
+                Comparator.comparing(
+                    ptn -> Conversions.fromByteBuffer(Types.IntegerType.get(), ptn.lowerBound())))
+            .map(
+                p ->
+                    Pair.of(
+                        (Integer)
+                            Conversions.fromByteBuffer(Types.IntegerType.get(), p.lowerBound()),
+                        (Integer)
+                            Conversions.fromByteBuffer(Types.IntegerType.get(), p.upperBound())))
+            .collect(Collectors.toList());
+
+    List<Integer> lowers = c3Boundaries.stream().map(t -> t.first()).collect(Collectors.toList());
+    List<Integer> uppers = c3Boundaries.stream().map(t -> t.second()).collect(Collectors.toList());
+
+    // With custom sorting, this looks like
+    // - manifest 1 -> [lower bound = 0, upper bound = 4]
+    // - manifest 2 -> [lower bound = 4, upper bound = 9]
+    // Without the custom sorting, each manifest tracks the full range of c3 upper/lower bounds.
+    // AKA they look like
+    // - manifest 1 -> [lower bound = 0, upper bound = 9]
+    // - manifest 2 -> [lower bound = 0, upper bound = 9]
+    // So the upper bound of the partitions tracked by the first file should be LEQ the lower bounds
+    // of the second. Etc
+    assertThat(uppers.get(0))
+        .withFailMessage(
```

Comment:
I think what you rather want here is using `.as(...)` because it will then show actual/expected values, whereas `.withFailMessage()` will only show this message without actual/expected